src/squeeze_xen.ml

```diff
@@ -207,7 +205,7 @@ module Domain = struct
          let process_one_watch xc xs (path, token) =
            if path = _introduceDomain || path = _releaseDomain
            then look_for_different_domains ()
-           else match List.filter (fun x -> x <> "") (Xstringext.String.split '/' path) with
+           else match List.filter (fun x -> x <> "") (Astring.String.cuts ~empty:false ~sep:"/" path) with
```

Comment:
`String.split` is equivalent to `Astring.String.cuts` (with the default `~empty:true`), if you already say `~empty:false` you don't need to filter the list (I've seen this redundant filter in rrdd as well, I will fix it there). Please replace `List.filter (fun x -> x <> "") (Astring.String.cuts ~empty:false ~sep:"/" path)` with `Astring.String.cuts ~empty:false ~sep:"/" path`